sui-execution/latest/sui-move-natives/src/lib.rs

```diff
@@ -658,6 +674,25 @@ pub fn all_natives(silent: bool) -> NativeFunctionTable {
             make_native!(test_utils::create_one_time_witness),
         ),
     ];
+
+    // Hack: Only add the check_zklogin_id_internal and check_zklogin_iss_internal native functions
+    // if their cost base is set in the protocol config which happened from protocol version 26.
+    if protocol_config
+        .check_zklogin_id_cost_base_as_option()
+        .is_some()
+    {
+        sui_framework_natives.push((
+            "zklogin",
+            "check_zklogin_id_internal",
+            make_native!(zklogin::check_zklogin_id_internal),
+        ));
+        sui_framework_natives.push((
+            "zklogin",
+            "check_zklogin_iss_internal",
+            make_native!(zklogin::check_zklogin_iss_internal),
+        ));
```

Comment:
cc @amnn - not sure if this is really necessary when adding native methods. it seems that nothing bad would happen if adding the natives to a framework that doesn't support them, since they would simply remain uncallable.

Reply:
@jonas-lj reached out to me today because there was an issue with the native function set-up here.  The issue was with assuming that there would be `Some(...)` cost parameters for these functions in the protocol config, which may not be true at protocol config versions < 26 that are still using the latest execution version (protocol versions 18 and above).

The way this code is currently architected, you're right, we no longer need to gate adding the native functions based on the protocol config, because the cost in the `NativesCostTable` is now `Option`-al and only gets checked when you call the native function.

I think it's fine to do it this way, and then we can ditch the conditional here.  The alternative is to associate the cost param with the native at the time it is bound (in which case you would need to keep the conditional here, but can remove the invariant violation check in the body of the native function, because you are guaranteed to get a valid cost param).